File: app/web/src/types/provider-status.ts
export interface ProviderStatus {
  provider: string;
  status: "CONNECTED" | "DEGRADED" | "UNAVAILABLE" | "DISABLED";
  lastSuccessfulUpdate: string | null;
  dataType: string | null;
  message: string | null;
}

export interface ProviderStatusResponse {
  stocks: ProviderStatus;
  mutualFunds: ProviderStatus;
}

File: app/api/src/main/java/finadvisor/marketdata/ProviderStatus.java
package finadvisor.marketdata;

import java.time.Instant;

/** Public-safe snapshot of a provider's health - never includes credentials or internal error detail. */
public record ProviderStatus(
        String provider,
        ProviderConnectionStatus status,
        Instant lastSuccessfulUpdate,
        DataType dataType,
        String message
) {
}
